02_plotting-mutation-counts.Rmd

```diff
@@ -0,0 +1,265 @@
+(Thankfully, paths like this *do* work on Windows now!) Using the separated list of elements also makes it easier to join partial paths together.
+We won't have to worry about whether a partial path ends with a separator slash or not; `file.path()` will add it as needed!
+
+```{r inputs}
+# Set the directory where the counts files are found
+counts_dir <- here::here("analysis", "gene_counts")
+
+# File paths to the counts files for LGAT and medulloblastoma
+lgat_path <- file.path(counts_dir, "LGAT_gene-mutations.tsv")
+mb_path <- file.path(counts_dir, "Medulloblastoma_gene-mutations.tsv")
+
+# Reference directory & FLAGS (frequently mutated genes)
+ref_dir <- here::here("data", "reference")
+flags_path <- file.path(ref_dir, "FLAGS.txt")
+```
+
+```{r outputs}
+# set the directory to save output files
+plots_dir <- here::here("plots", "gene_counts")
+
+# create the plots directory if it does not exist
+if (!dir.exists(plots_dir)){
+  dir.create(plots_dir, recursive = TRUE)
+}
+```
+
+### Define functions
+
+In this section we define any functions that we will use later in the script.
+While we could define functions later, it is nice to have them grouped together near the top of the notebook for at least a few reasons: - The first is that we will always know where to look for function definitions.
+Patterns and conventions make it easier for us to know what to expect when we come back to a file after a while away and may need to make changes - Another reason is that if the function definitions are at the top, we can use them anywhere below, without worrying about invoking a function before it is defined.
+Finally, having the functions grouped together will make it easier if we decide to move them to an external file: - If the set of functions gets large, or if we want to use some functions across multiple notebooks, we can create a separate `.R` file (or files) and then use the `source()` command to load those functions into this notebook (or others) all at once.
+
+In this case, we will only have one function, which will make a plot of the number of samples with mutations for each gene.
+We'll give it some options to allow filtering to a minimum number of samples mutated, and highlighting of particular genes of interest.
+
+Importantly, we will be sure to include some comments with documentation of the arguments for the function, as well as a description of what the function will return.
+
+To make this a bit more structured we will use a format called roxygen for these comments, which would be useful for automatically generating documentation if we were creating an R package.
+We aren't doing that here, but it is still a nice enough convention, and RStudio has an option in the `Code` menu called `Insert Roxygen Skeleton` that can save us a bunch of typing and create the outline of what we need in a single step.
```

Comment:
Something needs to be cleaned up for whether we use roxygen or not, because is it clearly used below?